src/codegen/templates/property_setter_template.py.mako

```diff
@@ -22,47 +23,20 @@
         val = ${attribute.ctypes_data_type}(val)
     %endif
 \
+## Script interpreter call.
 <%
-        argtypes = []
-        for handle_parameter in attribute.handle_parameters:
-            if handle_parameter.ctypes_data_type == 'ctypes.c_char_p':
-                argtypes.append('ctypes_byte_str')
-            else:
-                argtypes.append(handle_parameter.ctypes_data_type)
-        if (attribute.is_list and attribute.ctypes_data_type != 'ctypes.c_char_p' and
-                attribute.bitfield_enum is None):
-            argtypes.append("wrapped_ndpointer(dtype={}, flags=('C','W'))"
-                            .format(attribute.ctypes_data_type))
-        elif attribute.ctypes_data_type == 'ctypes.c_char_p':
-            argtypes.append('ctypes_byte_str')
-        else:
-            argtypes.append(attribute.ctypes_data_type)
-        if attribute.has_explicit_write_buffer_size:
-            argtypes.append('ctypes.c_uint')
-    %>\
-    ## When the length of the function name is too long, it will be wrapped to the next line
-    %if len(attribute.c_function_name) < 33:
-        cfunc = lib_importer.${attribute.get_lib_importer_type()}.DAQmxSet${attribute.c_function_name}
-    %else:
-        cfunc = (lib_importer.${attribute.get_lib_importer_type()}.
-                 DAQmxSet${attribute.c_function_name})
-    %endif
-        if cfunc.argtypes is None:
-            with cfunc.arglock:
-                if cfunc.argtypes is None:
-                    cfunc.argtypes = [
-                        ${', '.join(argtypes) | wrap(initial_indent=24)}]
-\
-## Script function call.
-<%
+        mapped_func_type = get_mapped_attribute_function_type(attribute)
+        generic_attribute_func = get_generic_attribute_function_name(attribute) + "_" + mapped_func_type
         function_call_args = []
         for handle_parameter in attribute.handle_parameters:
             function_call_args.append(handle_parameter.accessor)
+        if attribute.python_class_name == "Watchdog":
+            function_call_args.append("\"\"")
+        function_call_args.append(hex(attribute.id))
         function_call_args.append('val')
+
         if attribute.has_explicit_write_buffer_size:
             function_call_args.append('len(val)')
     %>\
-        error_code = cfunc(
-            ${', '.join(function_call_args) | wrap(initial_indent=12)})
-        check_for_error(error_code)
+        self._interpreter.set_${generic_attribute_func}(${', '.join(function_call_args) | wrap(12)})
```

Comment:
This makes the 1st line extra long because wrap(12) doesn't account for `self._interpreter.set_${generic_attribute_func}(`. Drop the args to the next line.
```suggestion
        self._interpreter.set_${generic_attribute_func}(
            ${', '.join(function_call_args) | wrap(12)})
```

Reply:
Ok. As discussed in this PR comment https://github.com/ni/nidaqmx-python/pull/298#discussion_r1178559224, I have removed `wrap()` since the length of the `setter()` properties are always <68 whitespaces.